src/infrastructure/content/ContentCommunicator.java

```diff
@@ -1,24 +1,248 @@
 package infrastructure.content;
 
+import infrastructure.validation.logger.ILogger;
+import infrastructure.validation.logger.LogLevel;
+import infrastructure.validation.logger.LoggerFactory;
+import infrastructure.validation.logger.ModuleID;
+import java.util.HashMap;
+import networking.CommunicatorFactory;
+import networking.ICommunicator;
+import networking.INotificationHandler;
+import org.json.JSONObject;
+
+/**
+ * This class ContentCommunicator implements the IContentCommunicator interface and all the 
+ * methods inside it. The ideal purpose of this class is to initialise any new client, send 
+ * message from UI to networking module, handle client exit with respect to content module
+ * and handle the subscribers of content module.
+ * 
+ * @author Badal Kumar (111701008)
+ */
 public class ContentCommunicator implements IContentCommunicator{
-    
+	/**
+	 * This variable will store the username/userid of client as a String.
+	 */
+	private static String userName;
+	
+	/**
+	 * This variable will store the image of client in String format.
+	 */
+	private String userImage;
+	
+	/**
+	 * This variable will store the server's ipAddress as a String.
+	 */
+	private String serverIPAddress;
+	
+	/**
+	 * Creating a HashMap to store userName and userImage of all active clients of a particular board.
+	 * Both userName and userImage are of String type
+	 */
+	private static HashMap<String, String> imageMap = new HashMap<String, String>();
+	
+	/**
+	 * Creating a HashMap to store the identifier and handler of subscriber of this module,
+	 * identifier is of type String while handler is of type IContentNotificationHandler.
+	 */
+	private static HashMap<String, IContentNotificationHandler> handlerMap = new HashMap<String, IContentNotificationHandler>();
+	
+	/**
+	 * An instance of ServerPort class is created and stored the variable serverPort.
+	 */
+	private ServerPort serverPort = new ServerPort();
+	/**
+	 * The variable port will store the port number of the board which the client is accessing.
+	 */
+	private int port = serverPort.getPort();
+	
+	/**
+	 * This variable will store the communicator of networking module to send data over the network.
+	 */
+	private ICommunicator communicator = CommunicatorFactory.getCommunicator(port);
+	
+	/**
+	 * logger is the instance of the class which implements ILogger interface.
+	 */
+	private static ILogger logger = LoggerFactory.getLoggerInstance();
+	
+	/**
+	 * This variable will store the String which will be passed to log method of logger
+	 */
+	private static String logMessage;
+	

```

Comment:
This need not be static

Reply:
it can't be used inside static method then.

Reply:
This is for the logger calls inside the getter and setter function, right. Logging inside getters and setters is not really required but it is already done, so let it be